internal/releaseapi/client.go

```diff
@@ -162,12 +162,11 @@ func (c *Client) downloadBuild(build Build, checkSha256Sum string) (string, erro
 	log.Printf("dowloading release archive from %s", build.URL)
 
 	zipFile, zipLength, err := c.downloadReleaseArchive(build)
-	defer os.Remove(zipFile.Name())
-	defer zipFile.Close()
-
 	if err != nil {
 		return "", err
 	}
+	defer os.Remove(zipFile.Name())
+	defer zipFile.Close()
 
 	f, err := os.Open(zipFile.Name())
 	if err != nil {
```

Comment:
nitpick, I prefer more whitespace

```suggestion
	
	defer os.Remove(zipFile.Name())
	defer zipFile.Close()

	f, err := os.Open(zipFile.Name())
	
	if err != nil {
```